src/nrnpython/nrnpy_hoc.cpp

```diff
@@ -2613,19 +2613,17 @@ static void nrnpy_store_savestate_(char** save_data, uint64_t* save_data_size) {
     if (store_savestate_) {
         // call store_savestate_ with no arguments to get a byte array that we can write out
         auto args = nb::steal(PyTuple_New(0));
-        PyObject* result = PyObject_CallObject(store_savestate_, args.ptr());
-        Py_INCREF(result);
-        if (result == NULL) {
+        auto result = nb::borrow(PyObject_CallObject(store_savestate_, args.ptr()));
```

Comment:
[PyObject_CallObject](https://docs.python.org/3/c-api/call.html#c.PyObject_CallObject) returns a new reference. Do we want to leak the object? (I see it used to INCREF, so my question is more about the original implementation)

Reply:
Correct, this was extracted as a bug fix:
https://github.com/neuronsimulator/nrn/pull/3229

(and now there's even a PR for the fix.)

Reply:
and merged ... and soon reworked.